beps/0002-dynamic-frontend-plugins/README.md

```diff
@@ -154,12 +154,386 @@ Each plugin is required to provide manifest file (metadata) in predefined format
 
 Plugin discovery is a pre-requisite for Plugin registry. This should be responsible for scanning for available plugins and generating/modifying the plugin registry to always keep it up to date.
 
-## Design Details
+### CLI
+
+The following sections contains proposal of new CLI scripts to support the dynamic frontend plugins.
+
+#### Static and Dynamic Plugin Build Differences
+
+The major difference between static and dynamic plugins lies in their dependencies and how they are "installed" into a Backstage instance and whether they are bundled into final JS output during the build.
+
+While static plugins can define their dependencies using the dependencies field inside `package.json`, dynamic (module federation-based) plugins can't.
+
+A dynamic plugin requires all dependencies to be present in the build output. Module sharing is then used to optimize the "output size" at runtime. If a shared package is not found in the browser scope, or if the version range does not match the requirement, it falls back and loads additional JS assets from the "private" build and adds them to the scope.
+
+Webpack is also not the best tool for building traditional NPM packages. Dependencies can be externalized, but Webpack brings additional code into the bundles. Using Webpack external dependencies also conflicts with module federation shared modules. Externals will not be included even as a shared module fallback. As a result, such shared module can't be used because the dependency will not be defined in the init container.
+
+If the vision is to still support "static" plugins installed via NPM, it makes sense to keep the static and dynamic plugin builds separate and build two different bundles. Both can still be published as a single NPM package (not recommended), but to achieve optimal performance for both, there cannot be a shared output.
+
+#### Dynamic export CLI
+
+```sh
+backstage-cli plugin export-dynamic
+```
+
+Similar to the current build command, instead of producing static plugin output, it builds a dynamic plugin using Webpack and module federation plugins.
+
+**Arguments**
+
+- _path_ (optional)
+  - path to the plugin `package.json`
+  - defaults to `cwd`
+- _export-name_
+  - Export name from where the plugin root is accessible
+  - defaults to `pluginEntry`
+- _cwd_
+  - current working directory of the script
+  - defaults to current `cwd` value
```

Comment:
I think we can have all of these hardcoded tbh. The script always targets `cwd`, that's how the the existing commands work. The path is always `package.json` or defined inside it. The export is always the default one, which is already part of the new frontend and backend systems.